### PR TITLE
xml ingestion and bug fixes

### PR DESCRIPTION
**Pull request by mhdkassir**
## Summary
some improvements on the XML ingestion
- use the specified schema when loading data 
- add a text based privacy job for XML files with complex structure

Also some bug fixing:
- _comet_input_file_name_ missing for Position files ingestion
- apply postsql queries on the accepted DataFrame
- error on updating BQ table schema 

**PR Type: Bug Fix & Feature**

**Status: Ready to review**

**Breaking change? No**

### How has this been tested?
Unit Tests
End to end testing of XML ingestion

## Contributor checklist:
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [ ] I have updated the Release notes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala [hunk 1]
```diff
@@ -240,22 +242,28 @@ trait IngestionJob extends SparkJob {
       .getOrElse(finalAcceptedDF)
 
     logger.info("Merged Dataframe Schema")
-    mergedDF.printSchema()
+    val finalMergedDf = schema.postsql match {
+      case Some(req) =>
+        mergedDF.createOrReplaceTempView(s"${schema.name}_tbl")
```
[the remark this answers is not in the document]
Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala:247]: replace comet_table

File: src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala [hunk 2]
```diff
@@ -319,9 +327,9 @@ trait IngestionJob extends SparkJob {
             metadata.getWrite(),
             schema.merge.exists(_.key.nonEmpty)
           )
-          val tableSchema = schema.mergedMetadata(domain.metadata).getFormat() match {
-            case Format.XML => None
-            case _          => Some(schema.bqSchema(schemaHandler))
+          val tableSchema = schema.postsql match {
+            case Some(_) => Some(BigQueryUtils.bqSchema(mergedDF.schema))
```
[the remark this answers is not in the document]
Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala:333]: well spotted, done

File: src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala]: done

File: src/main/scala/com/ebiznext/comet/schema/model/Schema.scala [hunk 3]
```diff
@@ -67,7 +67,7 @@ case class Schema(
   merge: Option[MergeOptions],
   comment: Option[String],
   presql: Option[List[String]],
-  postsql: Option[List[String]],
+  postsql: Option[String] = None,
```
[the remark this answers is not in the document]
Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/schema/model/Schema.scala]: done

File: src/main/scala/com/ebiznext/comet/job/ingest/XmlIngestionJob.scala [hunk 4]
```diff
@@ -89,51 +90,38 @@ class XmlIngestionJob(
...
-      }
+        val allRejected = rejectedRDD.union(koRDD)
+        saveRejected(allRejected)
+        val transformedAcceptedDF = session.createDataFrame(okRDD, appliedSchema)
+        saveAccepted(transformedAcceptedDF)
+        (allRejected, okRDD)
     }
   }
+
+  override def name: String = "XMLJob"
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/job/ingest/XmlIngestionJob.scala:126]: Shouldn't we add a domain name / schema name  to the job name ?
Such as it is done DsvIngestionJob s"""${domain.name}-${schema.name}-${path.headOption.map(_.getName).mkString(",")}"""

Comment (mhdkassir) [reply src/main/scala/com/ebiznext/comet/job/ingest/XmlIngestionJob.scala]: yes! done

**Review by hayssams — APPROVED** [review]
LGTM

**Pull request merged**